server/game/cards/events/02/trialbycombat.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../../drawcard.js');
+const ApplyClaim = require('../../../gamesteps/challenge/applyclaim.js');
+
+class TrialByCombat extends DrawCard {
+    setupCardAbilities() {
+        this.interrupt({
+            when: {
+                onClaimApplied: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    challenge.attackingPlayer === this.controller &&
```

Comment:
True – I guess it's a trade-off, having the interrupt on both players intrigue challenge claim would probably cause more annoyance.